### PR TITLE
Update `match` examples

### PR DESCRIPTION
**Pull request by camiinthisthang**
## Description
Updated match examples to be more intuitive and helpful for the reader, replacing function and variable names like foo, bar, etc. 

## Checklist

- [X] I have linked to any relevant issues.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [X] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [X] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [X] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

**Comment by camiinthisthang** [comment]
Linter is failing due to [this open issue](https://github.com/FuelLabs/sway/issues/4092). There's a PR open to fix what's causing the linter to fail

**Comment by kayagokalp** [comment]
@camiinthisthang #4093 is merged 🎉, this should fix the CI. Merging master to get that change here

File: examples/match_statements/src/main.sw [hunk 1]
```diff
@@ -1,45 +1,48 @@
...
-    let x = 5;
-
-    // Match as an expression.
-    let a = match 8 {
-        7 => 4,
-        9 => 5,
-        8 => 6,
-        _ => 100,
+fn main(num: u64) -> u64 {
+// Match as an expression
```
Comment (jtriley-eth) [inline examples/match_statements/src/main.sw:12]: Should these comments be indented to match the current block? ie:
```rust
// this:
fn main(num: u64) -> u64 {
    // Match as an expression
    let isEven = match num % 2 {
        0 => true,
        _ => false,
    };
}

// as opposed to this:
fn main(num: u64) -> u64 {
// Match as an expression
    let isEven = match num % 2 {
        0 => true,
        _ => false,
    };
}
```

**Review by jtriley-eth — COMMENTED** [review]
One suggestion and one question, otherwise lgtm!

File: examples/match_statements/src/main.sw (hunk 1 above)
Comment (camiinthisthang) [reply examples/match_statements/src/main.sw]: fixed, thank you!

File: docs/book/src/basics/control_flow.md [hunk 2]
```diff
@@ -32,7 +32,20 @@ Note that all branches of the `if` expression must return a value of the same ty
 
 ### `match` expressions
 
-Sway supports advanced pattern matching through exhaustive `match` expressions.
+Sway supports advanced pattern matching through exhaustive `match` expressions. Unlike an `if` statement, a match statement asserts at compile time all possible patterns have been matched. If you don't handle all cases, you'll get a warning from the compiler indicating that you need to add a `_`, a catch-all pattern that matches any value and is used when you want to specify a default action.
```
Comment (mohammadfawaz) [inline docs/book/src/basics/control_flow.md:35]:
```suggestion
Sway supports advanced pattern matching through exhaustive `match` expressions. Unlike an `if` statement, a `match` expression asserts **at compile** time that all possible patterns have been matched. If you don't handle all the patterns, you will get compiler error indicating that your `match` expression is non-exhaustive.
```

File: docs/book/src/basics/control_flow.md [hunk 3]
```diff
@@ -32,7 +32,20 @@ Note that all branches of the `if` expression must return a value of the same ty
...
+
+The basic syntax of a `match` statement:
```
Comment (mohammadfawaz) [inline docs/book/src/basics/control_flow.md:37]:
```suggestion
The basic syntax of a `match` statement is as follows:
```

File: docs/book/src/basics/control_flow.md [hunk 4]
```diff
@@ -32,7 +32,20 @@ Note that all branches of the `if` expression must return a value of the same ty
...
+
+```sway
+match expression {
```
Comment (mohammadfawaz) [inline docs/book/src/basics/control_flow.md:40]: Just to emphasize that it is possible to return a value from a `match` expression.
```suggestion
let result = match expression {
```

**Review by mohammadfawaz — APPROVED** [review]
Excellent, thanks!

**Review by anton-trunov — APPROVED** [review]

**Pull request merged**